src/filing/actions/getFilingPeriodOptions.js

```diff
@@ -0,0 +1,125 @@
+import { splitYearQuarter } from "../api/utils"
+import { afterFilingPeriod, withinFilingPeriod } from "../utils/date"
+import receivePeriodOptions from "./receivePeriodOptions"
+import requestPeriodOptions from "./requestPeriodOptions"
+import fetchAllInstitutions from './fetchAllInstitutions'
+
+const QUARTERS = ["Q1", "Q2", "Q3"]
+
+export default function getFilingPeriodOptions(
+  institutions,
+  filingPeriods,
+  filingQuarters,
+  filingQuartersLate
+) {
+  let periodOptions = generatePeriodOptions(
+    filingPeriods,
+    filingQuarters,
+    filingQuartersLate
+  )
+
+  // Only need to check years with quarterly options
+  const yearFilterCandidates = new Set(
+    periodOptions
+      .filter((po) => po.indexOf("-Q") > -1)
+      .map((po) => splitYearQuarter(po)[0])
+  )
+
+  return (dispatch) => {
+    const optionDisplayMap = {}
+
+    dispatch(requestPeriodOptions)
+
+    fetchAllInstitutions(yearFilterCandidates, institutions).then((jsons) => {
+      jsons.forEach((json) => {
+        // Response object (has URL) vs Institution details object
+        let year = json.url
+          ? +json.url.split("/").slice(-1)
+          : json.institution.activityYear
+
+        // Any quarterly filer = should display year
+        if (!json.status)
+          optionDisplayMap[year] =
+            json.institution.quarterlyFiler || optionDisplayMap[year]
+      })
+
+      dispatch(
+        receivePeriodOptions(filterOptions(periodOptions, optionDisplayMap))
+      )
+    })
+  }
+}
+
+/*********************************************************
+********************* Service Methods ********************
+**********************************************************/
+
+/**
+ * Determine if a filingPeriod is currently open or has passed
+ * @param {String} period filingPeriod
+ * @param {Object} dates start/end dates for filing
+ * @param {Object} datesLate start/end dates for late filing
+ */
+function shouldAddPeriod(period, dates, datesLate) {
+  return (
+    withinFilingPeriod(period, dates) ||
+    withinFilingPeriod(period, datesLate) ||
+    afterFilingPeriod(period, dates) ||
+    afterFilingPeriod(period, datesLate)
+  )
+}
+
+/**
+ * Generate a list of viable filingPeriods
+ * @param {Array} periods List of open filingPeriods
+ * @param {Object} dates start/end dates for filing
+ * @param {Object} datesLate start/end dates for late filing
+ */
+function generatePeriodOptions(periods, dates, datesLate) {
```

Comment:
Based on feedback during standup, all available filing periods will remain the config, so this can be greatly simplified or even removed in favor of using the config.filingPeriods as the basis for which periods to consider.